File: OOP/Aula_07-Herenca/AtualizadorDeContas.java

public class AtualizadorDeContas {

	private double selic;
	private double saldoTotal = 0;

	// Getters
	public double getSelic()			{ return this.selic; }
	public double getSaldoTotal()			{ return this.saldoTotal; }

	// Setters
	public void setSelic(double selic_p)		{ this.selic = selic_p; }
	public void setSaldoTotal(double total_p)	{ this.saldoTotal = total_p; }

	// Constructors
	public AtualizadorDeContas (double selic) {
		this.setSelic (selic);
	}

	public void rodar(Conta c) {

		System.out.println("Saldo Anterior: " + c.getSaldo());

		c.atualizar(this.getSelic());

		System.out.println ("Saldo Atualizado: " + c.getSaldo() + "\n");

		this.setSaldoTotal (this.getSaldoTotal() + c.getSaldo());
	}
}

File: OOP/Aula_08_Classes_Abstratas/Att_01-05/AtualizadorDeContas.cs
using Entities;
public class AtualizadorDeContas
{
	private double selic;
	private double saldoTotal;

	public double Selic
	{
		get { return selic; }
		set { selic = value; }
	}
	public double SaldoTotal
	{
		get { return saldoTotal; } 
		set { saldoTotal = value; }
	}

	// Constructors
	public AtualizadorDeContas (double _selic)
	{
		selic = _selic;
	}

	public void rodar(Conta _conta)
	{
		Console.WriteLine ("Saldo Anterior: " + _conta.Saldo);
		_conta.Atualizar(selic);

		Console.WriteLine ("Saldo Atualizado: " + _conta.Saldo + "\n");
		saldoTotal += _conta.Saldo;
	}
}
